### DISCUSSION TITLE
Converting text to a path creates multiple paths

### DISCUSSION THREAD

**Discussion by emnrdl** [mozman/ezdxf#1201] [Q&A]
When I convert to text entity to a path its create mutliple paths but when I convert to text in corel its create one single path. I want to create a one continius path like corel did how can i achive this ?

```Python
for i, char in enumerate(text):
                        # Get the relative width of the character
                        char_width = new_char_widths[i]  # Use default width if not found
                        
                        # Calculate the angular space for this character (arc length corresponding to the character width)
                        char_arc_length = char_width * bottom_text_height
                        char_angle = (char_arc_length / (2 * math.pi * radius)) * 360  # Convert arc length to angle

                        # Calculate the angle for the current character's position
                        angle_radians = degrees_to_radians(current_angle)
                        
                        # Calculate the position on the circle
                        x = center[0] + radius * math.cos(angle_radians)
                        y = center[1] + radius * math.sin(angle_radians)
                        
                        # Create a text entity with the custom font style
                        text_entity = msp.add_text(char, dxfattribs={'height': bottom_text_height, 'style': 'CustomFont'})
                        
                        # Set text position using
                        text_entity.set_placement(p1=(x, y), align=ezdxf.enums.TextEntityAlignment.MIDDLE_CENTER)
                        
                        # Rotate the text to align with the circle using dxf.rotation
                        text_entity.dxf.rotation = current_angle + 90  # Add 90 degrees to align tangent to the circle
                        
                        
                        path_object = ezdxf.addons.text2path.make_paths_from_entity(text_entity)
                        path.render_splines_and_polylines(msp, path_object)
```

![corel](https://github.com/user-attachments/assets/5a1f146e-abbe-489a-86e5-03906eed3017)
![ezdxf](https://github.com/user-attachments/assets/1758f164-09d8-4a8a-a5aa-f203f7e8c538)

**Comment by mozman** [comment]
DXF has no entity which combines linear edges and spline edges as a single polyline. Multi-polygons or -polylines are also not supported.

- Render the text-paths as a solid HATCH which is a single entity but you get only the filling without the border line.
- Render the text-paths a LWPOLYLINES, which creates multiple closed polylines with approximated splines.

**Reply by mozman** [reply]
You can also render the text-paths into a new BLOCK.

**Reply by emnrdl** [reply]
I'm trying to achieve this result. Left one is made from corel draw right one is exported from ezdxf. I add line thickness both of them. There are cut parts on the right one. How can i get this result like left one. If you can give an idea it would be great.
![compare](https://github.com/user-attachments/assets/995f2a9e-a594-4668-9f31-087a8c43b84f)
